lib/ansiblelint/utils.py

```diff
@@ -599,55 +599,109 @@ def get_first_cmd_arg(task):
-    """ Uses ruamel.yaml to parse comments then adds a
-        skipped_rules list to the task (or meta yaml block)
+    """Append 'skipped_rules' to individual tasks or single metadata block.
+
+    For a file, uses 2nd parser (ruamel.yaml) to pull comments out of
+    yaml subsets, check for '# noqa' skipped rules, and append any skips to the
+    original parser (pyyaml) data relied on by remainder of ansible-lint.
+
+    :param pyyaml_data: file text parsed via ansible and pyyaml.
+    :param file_text: raw file text.
+    :param file_type: type of file: tasks, handlers or meta.
+    :returns: original pyyaml_data altered with a 'skipped_rules' list added
+    to individual tasks, or added to the single metadata block.
     """
 
-    PLAYBOOK_TASK_KEYWORDS = [
-        'tasks',
-        'pre_tasks',
-        'post_tasks',
-        'handlers',
-    ]
+    try:
+        yaml_skip = _append_skipped_rules(pyyaml_data, file_text, file_type)
+    except Exception as exc:
+        # Notify user of skip error, do not stop, do not change exit code
+        print('Error trying to append skipped rules: {}'.format(repr(exc)))
+        return pyyaml_data
+    return yaml_skip
 
+
+def _append_skipped_rules(pyyaml_data, file_text, file_type):
+    # parse file text using 2nd parser library
     yaml = ruamel.yaml.YAML()
     ruamel_data = yaml.load(file_text)
 
+    # if file type is 'meta', add skips to metadata block and return
+    if file_type == 'meta':
+        skipped_rules = _get_rule_skips_from_yaml(ruamel_data)
+        if skipped_rules:
+            pyyaml_data[0]['skipped_rules'] = skipped_rules
+        return pyyaml_data
+
+    # create list of blocks of tasks or nested tasks
     if file_type in ('tasks', 'handlers'):
-        ruamel_tasks = ruamel_data
-        pyyaml_tasks = pyyaml_data
+        ruamel_task_blocks = ruamel_data
+        pyyaml_task_blocks = pyyaml_data
     elif file_type == 'playbook':
         try:
-            ruamel_tasks = []
-            pyyaml_tasks = []
-            for ruamel_play, pyyaml_play in zip(ruamel_data, pyyaml_data):
-                for key in PLAYBOOK_TASK_KEYWORDS:
-                    ruamel_tasks.extend(ruamel_play.get(key, []))
-                    pyyaml_tasks.extend(pyyaml_play.get(key, []))
+            pyyaml_task_blocks = _get_task_blocks_from_playbook(pyyaml_data)
+            ruamel_task_blocks = _get_task_blocks_from_playbook(ruamel_data)
         except (AttributeError, TypeError):
+            # TODO(awcrosby): running ansible-lint on any .yml file will
+            # assume it is a playbook, check needs to be added higher in the
+            # call stack, and can remove this except
             return pyyaml_data
-    elif file_type == 'meta':
-        if not isinstance(pyyaml_data, list):
-            return pyyaml_data
-        ruamel_tasks = [ruamel_data]
-        pyyaml_tasks = pyyaml_data
     else:
-        return pyyaml_data
+        raise RuntimeError('Unexpected file type')
```

Comment:
This error will probably benefit from the context added:
```suggestion
        raise RuntimeError('Unexpected file type: {}'.format(file_type))
```